optaplanner-core/src/main/java/org/optaplanner/core/impl/score/stream/bi/NoneBiJoiner.java

```diff
@@ -22,23 +22,35 @@
 
 public final class NoneBiJoiner<A, B> extends AbstractBiJoiner<A, B> {
 
+    private static final Object[] EMPTY_OBJECT_ARRAY = new Object[0];
+    private static final JoinerType[] EMPTY_JOINER_ARRAY = new JoinerType[0];
+
     // ************************************************************************
     // Builders
     // ************************************************************************
 
+    @Override
+    public Function<A, Object> getLeftMapping(int joinerIndex) {
+        throw new UnsupportedOperationException();
```

Comment:
Can we change this to:
throw new UnsupportedOperationException("Impossible state: getLeftMapping is never called on a NoneBiJoiner.");

The "Impossible state" is our way of saying: we're theoretically a 100% sure that our code will never do this, so if you do see this, it's a bug and it's not your fault, user.